x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -155,11 +141,15 @@ public void init() {
             ),
             Set.of("foo")
         );
+        analyzer = makeAnalyzer("mapping-basic.json", enrichResolution);
+    }
 
-        analyzer = new Analyzer(
-            new AnalyzerContext(config, functionRegistry, getIndexResult, enrichResolution),
-            new Verifier(new Metrics())
-        );
+    private Analyzer makeAnalyzer(String mappingFileName, EnrichResolution enrichResolution) {
```

Comment:
This is required for the backported test; to avoid code duplication in `init`, I updated `init` as well to be the same as in main.